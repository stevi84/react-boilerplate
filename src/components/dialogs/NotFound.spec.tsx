import { describe, it, expect, vi } from 'vitest';
import { ShallowRenderer, createRenderer } from 'react-test-renderer/shallow';
import { NotFound } from './NotFound';

vi.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (str: string) => str,
    i18n: {
      language: 'de',
      changeLanguage: vi.fn(),
    },
  }),
}));

vi.mock('react-router-dom', () => ({ useNavigate: () => vi.fn() }));

vi.mock('../../hooks/UseNotifier', () => ({ useNotifier: vi.fn() }));

describe('NotFound', () => {
  it('should equal saved snapshot', () => {
    const renderer: ShallowRenderer = createRenderer();
    renderer.render(<NotFound />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
