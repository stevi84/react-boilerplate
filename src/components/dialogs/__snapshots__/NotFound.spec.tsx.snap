// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFound > should equal saved snapshot 1`] = `
<MainLayout>
  <ForwardRef(Paper)
    sx={
      {
        "margin": 1,
      }
    }
  >
    <ForwardRef(Grid)
      alignItems="center"
      direction="column"
      justifyContent="center"
    >
      <ForwardRef(Typography)
        color="primary"
        variant="h1"
      >
        404
      </ForwardRef(Typography)>
      <ForwardRef(Typography)
        color="textSecondary"
        variant="h4"
      >
        Not Found
      </ForwardRef(Typography)>
      <ForwardRef(Typography)
        color="textSecondary"
        variant="body1"
      >
        404
      </ForwardRef(Typography)>
    </ForwardRef(Grid)>
  </ForwardRef(Paper)>
</MainLayout>
`;
